From cd40bf093058321f010206fa3652cd3266b165f8 Mon Sep 17 00:00:00 2001
From: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Date: Tue, 9 Aug 2022 15:17:45 +0300
Subject: [PATCH 054/102] arm: s32cc: Get a reference to PCIe node before
 enabling it

Altering the PCIe node might affect the enabling operation as
the node might get shifted in device tree. This is the reason
a new reference to PCIe node should be obtained.

Issue: ALB-8357, ALB-9149
Upstream-Status: Pending 

Signed-off-by: Ghennadi Procopciuc <ghennadi.procopciuc@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 arch/arm/mach-s32/s32-cc/hwconfig_fixups.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c b/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
index 404988f001..bf50e2a09f 100644
--- a/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
+++ b/arch/arm/mach-s32/s32-cc/hwconfig_fixups.c
@@ -586,6 +586,12 @@ static int prepare_pcie_node(struct dts_node *root, int id)
 	if (ret)
 		return ret;
 
+	ret = node_by_alias(root, &node, PCIE_ALIAS_FMT, id);
+	if (ret) {
+		pr_err("Failed to get 'pcie%u' alias\n", id);
+		return ret;
+	}
+
 	ret = enable_node(&node);
 	if (ret) {
 		pr_err("Failed to enable PCIe%d\n", id);
-- 
2.17.1

